From 504c163114c127e13f57864d62b0b7c67a429538 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 10 May 2024 12:50:17 +0000
Subject: [PATCH] bootcount set0

---
 .../Application/L4TLauncher/L4TLauncher.c       | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
index 4b24f342..36769c8a 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
@@ -2905,14 +2905,15 @@ L4TLauncher (
               ErrorPrint(L"%a: Error while processing bootcount value\r\n", __FUNCTION__);
           }
           Print (L"%a: BootCountValue is %d\r\n", __FUNCTION__, BootCountValue);
-      } else if (2 == UpgradeAvailable) {
-	      ErrorPrint(L"%a: UpgradeAvailable set to 2, setting bootchain to 0\r\n", __FUNCTION__);
-	      Status = SetNextBootChain(0);
-	      if (EFI_ERROR (Status)) {
-	         ErrorPrint(L"Failed to SetNextBootChain to 0\r\n");
-	      }
-		ErrorPrint(L"Rebooting...\r\n\n");
-	      ResetCold();
+	  if (1 == BootCountValue) {
+	                ErrorPrint(L"%a: Bootcount1, setting bootchain to 0\r\n", __FUNCTION__);
+              Status = SetNextBootChain(0);
+              if (EFI_ERROR (Status)) {
+                 ErrorPrint(L"Failed to SetNextBootChain to 0\r\n");
+              }
+                ErrorPrint(L"Rebooting...\r\n\n");
+              ResetCold();
+	}
       }
 
       /* If boot limit is reached while upgrade_available=1, switch partitions */
-- 
2.17.1

